.github/workflows/test-unit.yml

```diff
@@ -52,16 +52,31 @@ jobs:
           sudo echo "127.0.0.1 etcd" | sudo tee -a /etc/hosts
 
       - name: Run tests
-        run: make tests-unit
         shell: bash
+        run: |
+          # ENVs can affect test caching, make sure to only run tests with expected ENVs
+          env -i \
+            TERM="$TERM" \
+            HOME="$HOME" \
+            PATH="$PATH" \
+            APPDATA="$APPDATA" \
+            DATADOG_ENABLED="false" \
+            UNIT_ETCD_ENABLED="$UNIT_ETCD_ENABLED" \
+            UNIT_ETCD_ENDPOINT="etcd:2379" \
+            UNIT_ETCD_USERNAME="root" \
+            UNIT_ETCD_PASSWORD="toor" \
+            UNIT_ETCD_NAMESPACE="unit" \
+            TEST_PARALLELISM="$TEST_PARALLELISM" \
+            TEST_PARALLELISM_PKG="$TEST_PARALLELISM_PKG" \
+            TEST_COVERAGE="$TEST_COVERAGE" \
+            TEST_CREATE_OUT_DIR="false" \
+            TEST_KBC_PROJECTS_LOCK_HOST="$TEST_KBC_PROJECTS_LOCK_HOST" \
+            TEST_KBC_PROJECTS_LOCK_PASSWORD="$TEST_KBC_PROJECTS_LOCK_PASSWORD" \
+            TEST_KBC_PROJECTS_FILE="$TEST_KBC_PROJECTS_FILE" \
+          make tests-unit
```

Comment:
> go-client change isn't strictly necessary

:heavy_check_mark: 

> So either one (go-client fix / env whitelist) would be sufficient?

:heavy_check_mark: 

`go-client` fix is needed for security reasons, even if the ENVs are not used, an auditor can ask why the CLI reads these ENVs. It can "steal" some AWS client session, more precisely, it looks like this.